actionpack/test/dispatch/cookies_test.rb

```diff
@@ -321,10 +322,11 @@ def test_setting_cookie_with_secure
   end
 
   def test_setting_cookie_with_secure_when_always_write_cookie_is_true
-    ActionDispatch::Cookies::CookieJar.any_instance.stubs(:always_write_cookie).returns(true)
+    @request.cookie_jar.always_write_cookie = true
     get :authenticate_with_secure
     assert_cookie_header "user_name=david; path=/; secure"
     assert_equal({"user_name" => "david"}, @response.cookies)
+    @request.cookie_jar.always_write_cookie = false
```

Comment:
There's a standard pattern of capturing the old value and then setting the variable back to that. Like this:

```ruby
def test_setting_cookie_with_secure_when_always_write_cookie_is_true
  old_cookie, @request.cookie_jar.always_write_cookie = @request.cookie_jar.always_write_cookie, true

  get :authenticate_with_secure
  # ...
ensure
  @request.cookie_jar.always_write_cookie = old_cookie
end
```